### PR TITLE
[pipeline] - Use the kubernetes agents

### PR DESCRIPTION
**Pull request by lazypower**
- shift from running in docker on the master node to using the kube
  builders dedicated to golang projects

### CODE REVIEW

**Review by hoanhan101 — APPROVED** [review]
lgtm! thanks!

File: .jenkins [hunk 1]
```diff
@@ -60,14 +50,11 @@ pipeline {
       }
       steps {
         script {
-          docker.image('vaporio/jenkins-agent-golang:latest').inside('-v /var/run/docker.sock:/var/run/docker.sock -v /usr/bin/docker:/usr/bin/docker -e GITHUB_TOKEN=$GITHUB_TOKEN --group-add docker') {
```
Comment (edaniszewski) [inline .jenkins:63]: So, this stage isn't being run in a golang container, so does that mean that it's running on the "golang-next" agent? If so, I'm just wondering what the difference is between running in a golang container vs. the golang-next agent, and when one should be preferred over the other

File: .jenkins [hunk 2]
```diff
@@ -2,7 +2,12 @@
 
 pipeline {
 
-  agent any
+  agent {
+    label 'golang-next'
+  }
+  environment {
+    GITHUB_TOKEN = credentials('vio-bot-gh-token')
+  }
```
Comment (edaniszewski) [inline .jenkins:9]: also, since we define the `GITHUB_TOKEN` here, do we still need it defined in the Tagged Release stage?

Comment (lazypower) [reply .jenkins:9]: nope :)

File: .jenkins (hunk 1 above)
Comment (lazypower) [reply .jenkins:63]: you shouldn't ever have to use the docker.image script hack now that this runs in [this agent](https://github.com/vapor-ware/jenkins-agents/blob/master/golang/Dockerfile) which has runtime tooling/support for what this container was doing on the build-master previously:

Not pictured: the docker socket / binary is volume mounted into the container :) 

There may be some bugs in this pipeline but i'm free to help sort them out.

Comment (edaniszewski) [reply .jenkins:63]: Good to know, thanks! I realize my question wasn't totally clear though, what I meant to ask was why the other stages are being run in `container('golang'){` where this is just being run on the golang agent directly

Comment (lazypower) [reply .jenkins:63]: laziness, and it should be updated. :smile:

**Review by edaniszewski — APPROVED** [review]

File: .jenkins (hunk 1 above)
Comment (lazypower) [reply .jenkins:63]: Bumped, and shuffled those credentials exposure around a bit based on the above comment. :+1: thanks for keeping me honest ;D

**Review by edaniszewski — APPROVED** [review]
lgtm! (:

**Pull request merged**